grommet/grommet
# Only set focus on FormField if it containsFocus in the DOM

#### What does this PR do?
This PR adjusts FormField setFocus to check if the FormField contains DOM focus. If true, it sets focus to true and applies focus styling. If false, it does not apply focus styling since another DOM element has focus.

This resolves a bug where the TextInput in Select + Search was displaying focus on both the FormField and the TextInput. See "Background context" below for more details.

with fix
<img width="274" alt="Screen Shot 2021-03-19 at 1 00 43 PM" src="https://user-images.githubusercontent.com/42451602/111830089-20751680-88b3-11eb-9fcf-ee54309f6f14.png">

without fix
<img width="323" alt="Screen Shot 2021-03-19 at 1 01 07 PM" src="https://user-images.githubusercontent.com/42451602/111830132-2cf96f00-88b3-11eb-9829-fb6e6ca3f729.png">

#### Where should the reviewer start?
Formfield.js
#### What testing has been done on this PR?
storyboook
#### How should this be manually tested?
When `select` is in `formfield` there should only be one focus
```
        <FormField>
          <Select
            size="medium"
            placeholder="Select single option"
            value={value}
            options={options}
            onChange={({ option }) => setValue(option)}
            onClose={() => setOptions(defaultOptions)}
            onSearch={text => {
              // The line below escapes regular expression special characters:
              // [ \ ^ $ . | ? * + ( )
              const escapedText = text.replace(/[-\\^$*+?.()|[\]{}]/g, '\\$&');

              // Create the regular expression with modified value which
              // handles escaping special characters. Without escaping special
              // characters, errors will appear in the console
              const exp = new RegExp(escapedText, 'i');
              setOptions(defaultOptions.filter(o => exp.test(o)));
            }}
          />
        </FormField>

```
#### Any background context you want to provide?

Select with Search applies focus onto the Search Input when the Select Drop opens. However, FormField is controlling its own focus internally and was not aware that the DOM focus had been placed on the search input in the select drop. This resulted in a focus indicator being placed on both the search input and the formfield (even though the Search Input was the active DOM element). By adding this check, we ensure that FormField only sets its focus to true when it also has focus in the DOM.

#### What are the relevant issues?
https://github.com/grommet/grommet-theme-hpe/issues/171
#### Screenshots (if appropriate)

#### Do the grommet docs need to be updated?
no
#### Should this PR be mentioned in the release notes?

#### Is this change backwards compatible or is it a breaking change?
backwards compatible
